### PR TITLE
Fix fft in according to the article source for legacy conversion

### PR DESCRIPTION
**Pull request by rikirenz**
This fix should allow to export the fft field in the right format

### CODE REVIEW

**Comment by rikirenz** [comment]
ready to be reviewed

File: tests/integration/workflows/test_arxiv_workflow.py [hunk 1]
```diff
@@ -392,3 +394,77 @@ def test_harvesting_arxiv_workflow_accepted(
         obj = workflow_object_class.get(obj_id)
         # It was accepted
         assert obj.status == ObjectStatus.COMPLETED
+
+        # Check the conversion for the legacy sys
+        marc_json = hep2marc.do(obj.data)
+        marcxml = legacy_export_as_marc(marc_json)
+        record = create_record(marcxml)
```
Comment (jacquerie) [inline tests/integration/workflows/test_arxiv_workflow.py]: This is not the best record to test, because in INSPIRE we don't have it, so the things that in the assertions are on the left actually come from the current implementation.

So this is only testing that we don't regress from here, not that we create records that are the same as we have on INSPIRE...

Comment (jacquerie) [inline tests/integration/workflows/test_arxiv_workflow.py]: Yeah, actually we can do even better that this: we store an XML fixture of the record how it is on INSPIRE, and we do a similar test (without checking at all the intermediate steps). Then the test compares the dictionary with MARC keys coming from this fixture and the dictionary with MARC keys coming through the workflow: in this way you don't have to type manually all these assertions, but can write just one.

**Comment by kaplun** [comment]
@rikirenz can you rebase properly this PR?

**Comment by kaplun** [comment]
This PR seems not to correspond to what the title say(!) :dancing_women:

File: inspirehep/modules/workflows/tasks/submission.py [hunk 2]
```diff
@@ -373,11 +374,20 @@ def prepare_files(obj, eng):
     if not obj.files:
         return
 
+    def _define_filename(paper, file_name):
+        """Define the filename in according to the article source"""
```
Comment (jacquerie) [inline inspirehep/modules/workflows/tasks/submission.py]: There's no need to add docstrings to internal helpers: you're their only user, so they don't need documentation.

File: inspirehep/modules/workflows/tasks/submission.py [hunk 3]
```diff
@@ -34,6 +34,7 @@
 
 from invenio_accounts.models import User
 
+from ....modules.workflows.tasks.actions import is_arxiv_paper
```
Comment (jacquerie) [inline inspirehep/modules/workflows/tasks/submission.py]: This is just `.actions`, so it can added to the import two lines below.

File: inspirehep/modules/workflows/tasks/submission.py [hunk 4]
```diff
@@ -373,11 +374,20 @@ def prepare_files(obj, eng):
     if not obj.files:
         return
 
+    def _define_filename(paper, file_name):
+        """Define the filename in according to the article source"""
+        if is_arxiv_paper(paper):
+            return 'arxiv:' + file_name
+
+        # Default return
```
Comment (jacquerie) [inline inspirehep/modules/workflows/tasks/submission.py]: You don't need to write these kind of comments. It's obvious that a return returns!

File: inspirehep/modules/workflows/tasks/submission.py [hunk 5]
```diff
@@ -389,7 +399,6 @@ def _get_fft(url, name):
         _get_fft(pdf_file_obj.obj.file.uri, name)
         for name, pdf_file_obj in pdf_file_objs
     ]
-
```
Comment (jacquerie) [inline inspirehep/modules/workflows/tasks/submission.py]: Leftover?

File: inspirehep/modules/workflows/tasks/submission.py (hunk 2 above)
Comment (kaplun) [inline inspirehep/modules/workflows/tasks/submission.py]: Still isn't it nice to describe what the subfunction does, so that you don't have to reparse it with your eyes everytime?

**Comment by rikirenz** [comment]
@jacquerie unit tests are coming :+1:

File: tests/unit/workflows/test_workflows_tasks_submission.py [hunk 6]
```diff
@@ -35,9 +41,51 @@ class DummyEng(object):
...
 def test_add_note_entry_does_not_add_value_that_is_already_present():
     obj = StubObj({'public_notes': [{'value': '*Temporary entry*'}]}, {'core': 'something'})
     eng = DummyEng()
 
     assert add_note_entry(obj, eng) is None
     assert {'public_notes': [{'value': '*Temporary entry*'}]} == obj.data
+
+
+@mock.patch('inspirehep.modules.workflows.tasks.submission.is_arxiv_paper',
+    side_effect=fake_is_arxiv_paper)
```
Comment (jacquerie) [inline tests/unit/workflows/test_workflows_tasks_submission.py]: If all you want to do is to return a value you can just `return_value=True` without having to define a mock function.

File: tests/unit/workflows/test_workflows_tasks_submission.py [hunk 7]
```diff
@@ -35,9 +41,51 @@ class DummyEng(object):
...
+        'fake_record': mock_file,
+        '': mock_file,
+        'fake_record1.pdf': None,
+    })
+
+    assert prepare_files(files_collection, eng) is None
+    assert {'url': '/code/fake_uri',
+            'filetype': '.pdf',
+            'docfile_type': 'INSPIRE-PUBLIC',
+            'filename': 'arxiv:fake_record'} == files_collection.data['fft'][0]
```
Comment (jacquerie) [inline tests/unit/workflows/test_workflows_tasks_submission.py]: Better (you assert more, you don't `IndexError`):
```python
assert files_collection.data['fft'] == [
    {
        ...
    }
]
```

File: inspirehep/dojson/common/base.py [hunk 8]
```diff
@@ -304,6 +305,7 @@ def fft2marc(self, key, value):
         'o': value.get('flag'),
         'd': value.get('description'),
         'n': value.get('filename'),
+        'f': value.get('filetype'),
```
Comment (jacquerie) [inline inspirehep/dojson/common/base.py]: These two are modifying another thing in DoJSON and need tests as well.

File: tests/unit/workflows/test_workflows_tasks_submission.py [hunk 9]
```diff
@@ -35,9 +41,51 @@ class DummyEng(object):
...
+
+        def __getitem__(self, item):
+            return self.data[item]
+
+    eng = DummyEng()
+    files_collection = MagicMock()
+    mock_file = MagicMock()
+
+    mock_file.name = MagicMock(return_value='fake_record')
+    mock_file.obj.file.uri = 'fake_uri'
```
Comment (jacquerie) [inline tests/unit/workflows/test_workflows_tasks_submission.py]: I understand what you are doing with the mocks here, but I wonder if these could be pulled up in the `MockObj` definition rather than defined per test.

File: tests/unit/workflows/test_workflows_tasks_submission.py [hunk 10]
```diff
@@ -35,9 +41,51 @@ class DummyEng(object):
...
+
+    files_collection.data = {}
+    files_collection.files = MockFilesCollectionIterator({
+        'fake_record.pdf': mock_file,
+        'fake_record.txt': mock_file,
+        'fake_record': mock_file,
+        '': mock_file,
+        'fake_record1.pdf': None,
+    })
```
Comment (jacquerie) [inline tests/unit/workflows/test_workflows_tasks_submission.py]: An added benefit of the modification above is that these test scenarios can be split in multiple tests, which achieves better clarity.

File: inspirehep/dojson/common/base.py (hunk 8 above)
Comment (jacquerie) [inline inspirehep/dojson/common/base.py]: Also these should be split in their own commit.

File: tests/unit/workflows/test_workflows_tasks_submission.py (hunk 6 above)
Comment (jacquerie) [inline tests/unit/workflows/test_workflows_tasks_submission.py]: BTW this is an example of Classicist vs Mockist testing: http://martinfowler.com/articles/mocksArentStubs.html#ClassicalAndMockistTesting.

File: tests/unit/workflows/test_workflows_tasks_submission.py (hunk 9 above)
Comment (jacquerie) [inline tests/unit/workflows/test_workflows_tasks_submission.py]: Yes, sorry, `StubObj`!

Comment (rikirenz) [inline tests/unit/workflows/test_workflows_tasks_submission.py]: Could you show me the classicist way to simulate this behavior:
`mock_file.obj.file.uri = 'fake_uri'`?

File: inspirehep/modules/workflows/tasks/submission.py (hunk 2 above)
Comment (david-caro) [inline inspirehep/modules/workflows/tasks/submission.py]: 👍 
bad code < bad code commented < 'good code' < 'good code' commented < good code that needs no comments

**Comment by kaplun** [comment]
What is the status of this PR? @david-caro are you taking it into account WRT the work you are currently doing on passing over the uploaded file?

**Comment by jacquerie** [comment]
> What is the status of this PR?

I'm currently merging it.
